pkg/install/install_test.go

```diff
@@ -120,3 +123,39 @@ func TestDeployARMTemplate(t *testing.T) {
 		})
 	}
 }
+
+func TestAddResourceProviderVersion(t *testing.T) {
+	ctx := context.Background()
+
+	clusterdoc := &api.OpenShiftClusterDocument{
+		Key: "test",
+		OpenShiftCluster: &api.OpenShiftCluster{
+			Properties: api.OpenShiftClusterProperties{},
+		},
+	}
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	openshiftClusters := mock_database.NewMockOpenShiftClusters(controller)
+	openshiftClusters.EXPECT().
+		PatchWithLease(gomock.Any(), clusterdoc.Key, gomock.Any()).
+		DoAndReturn(func(ctx context.Context, key string, f func(doc *api.OpenShiftClusterDocument) error) (*api.OpenShiftClusterDocument, error) {
+			err := f(clusterdoc)
+			return clusterdoc, err
+		})
+
+	i := &Installer{
+		doc: clusterdoc,
+		db:  openshiftClusters,
+	}
+	err := i.addResourceProviderVersion(ctx)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Check it was set to the correct value
+	if i.doc.OpenShiftCluster.Properties.ProvisionedBy != version.GitCommit {
+		t.Error("version was not added")
+	}
```

Comment:
I guess this needs to be moved inside the DoAndReturn.  Ick, that's non-obvious.  Evidently Gomock isn't great for mocking out the database layer, we probably need some sort of separate mock database implementation one day - that's #59.

Reply:
Yeah, I agree. I've had great success in the past with verified fakes, because then you can treat the code under test a bit more generically instead of having to prime all the responses yourself.

Reply:
I've come up with an alternative method of testing that it was changed in the committed version. What do you think?

Reply:
One nit remaining.  Yeah, that'll do; if you fancy taking on #59 subsequently and moving that code out into a fake database I'd be fine with that.

Reply:
Thanks!